src/main/java/iroha/validation/transactions/storage/impl/BasicTransactionProvider.java

```diff
@@ -0,0 +1,106 @@
+package iroha.validation.transactions.storage.impl;
+
+import com.google.common.base.Strings;
+import io.reactivex.Observable;
+import iroha.protocol.BlockOuterClass.Block;
+import iroha.protocol.Queries;
+import iroha.protocol.Queries.BlocksQuery;
+import iroha.protocol.TransactionOuterClass.Transaction;
+import iroha.validation.transactions.storage.TransactionProvider;
+import iroha.validation.util.ObservableRxList;
+import java.security.KeyPair;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import jp.co.soramitsu.iroha.java.BlocksQueryBuilder;
+import jp.co.soramitsu.iroha.java.IrohaAPI;
+import jp.co.soramitsu.iroha.java.Query;
+import jp.co.soramitsu.iroha.java.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BasicTransactionProvider implements TransactionProvider {
+
+  private static final Logger logger = LoggerFactory.getLogger(MQTransactionProvider.class);
```

Comment:
Why `getLogger(MQTransactionProvider.class)`? Is it correct? Do we use `MQTransactionProvider.class` on puprose?

Reply:
Thanks! Done